packages/vite/src/node/plugins/html.ts

```diff
@@ -218,6 +218,36 @@ function handleParseError(
   )
 }
 
+/**
+ * Move importmap to top of <head> with `transformIndexHtml`.
+ * Must be added after post plugins.
+ */
+export function htmlImportMapPlugin(): Plugin {
+  const importMapRE =
+    /[ \t]*<script[^>]*type\s*=\s*["']?importmap["']?[^>]*>.*?<\/script>/is
+  return {
+    name: 'vite:html-importmap',
+    transformIndexHtml(html) {
+      // HTML must have head for importmap
+      if (!headPrependInjectRE.test(html)) return
+
+      let importMap: string | undefined
+      html = html.replace(importMapRE, (match) => {
+        importMap = match
+        return ''
+      })
+      if (importMap) {
+        html = html.replace(
+          headPrependInjectRE,
+          (match) => `${match}\n${importMap}`
+        )
+      }
```

Comment:
Maybe it is better to move `importmap` after the first `<script type="module">`?
If there are tags that are intended to be as first as possible (e.g. `<link rel="preload">`), I think we should place them as-is.
That said, I'm not sure if this is what we should take into consideration.

Reply:
Updated to prepend before `<script type="module">` instead. It doesn't make much difference in dev though since `/@vite/client` is always injected as first in head. In build that would be different, but I don't think the inconsistency would affect anything in prod.